app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -263,14 +266,34 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
     )
   }
 
-  fun markHomeMenuCloseDrawer() {
-    binding.fragmentDrawerNavView.menu.getItem(
-      NavigationDrawerItem.HOME.ordinal
-    ).isChecked =
-      true
+  fun restoreLastCheckedMenuItem() {
+    if (previousMenuItemId != null) {
+      val lastCheckedMenuItemValue: Int = previousMenuItemId!!
+      if (lastCheckedMenuItemValue == 0) {
+        getFooterViewModel().isAdministratorControlsSelected.set(true)
+        uncheckAllMenuItemsWhenAdministratorControlsIsSelected()
+        binding.fragmentDrawerNavView.menu.getItem(
+          NavigationDrawerItem.SWITCH_PROFILE.ordinal
+        ).isChecked =
+          false
```

Comment:
This seems a bit hacky to me. The navigation drawer in general should not specialize its behavior for how it's used outside of the drawer. Given that we have a separate call path for the exit dialog flow, could we use a different method to specifically handle this case via ExitProfileDialogFragment?

Reply:
Tried modifying it a bit.

Also, as per your comment in the previous PR, I am adding all the possibilities that we need to cover - 

1. When the dialog is created from the Nav drawer then we need to restore the menu item (it can be either `Administrator Controls`  or one of the 5 menuItems).
2. When the dialog is created by pressing the back button from the HomeActivity then we don't want to restore any of the items.

So, if we consider two variables `x` and `y` where `x` denotes the restoration of MenuItem and `y` denotes the restoration of Administrator Controls then either both of them are false or exactly one of them is true.

Reply:
The case you described (either both false or one is true) exactly what a proto oneof enforces (or a sealed class with an 'other' child). Given we're already trying to use proto to model this, I think your latest oneof is much closer to the ideal state.

I think the presenter is closer to the ideal state, too, but the part I still don't quite understand is why we need to specialize for administrator controls. Leaving another comment to focus on a potential alternative.